spec/requests/api/v1/merchant/quantity_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+describe 'Merchant API' do
+  context 'Quantity Route' do
+    it 'should return the quantity' do
```

Comment:
@jmejia How can I stub this out?

Reply:
@patrickshobe I should add... feel free to combine the concepts from both examples and figure out what suits your needs.

These couple of lines should replace all of the factories and data setup in your request spec. Then you should still really do the setup in a model test and test the stubbed method there.

Reply:
@jmejia That's awesome!  

So I have the full test running in the model spec
```
describe 'class methods' do
      before(:each) do
        @merchant_1 = create(:merchant)
        @merchant_2 = create(:merchant)
 
        @customer = create(:customer)
 
        @item_1 = create(:item, merchant: @merchant_1)
        @item_2 = create(:item, merchant: @merchant_2)
 
        @invoice_1 = create(:invoice, customer: @customer, merchant: @merchant_1)
        @invoice_2 = create(:invoice, customer: @customer, merchant: @merchant_1)
        @invoice_3 = create(:invoice, customer: @customer, merchant: @merchant_2)
 
        @invoice_item_1 = create(:invoice_item, item: @item_1, invoice: @invoice_1)
        @invoice_item_2 = create(:invoice_item, item: @item_1, invoice: @invoice_1)
        @invoice_item_3 = create(:invoice_item, item: @item_2, invoice: @invoice_3)
 
        @transaction_1 = create(:transaction, invoice: @invoice_1, result: "success")
        @transaction_2 = create(:transaction, invoice: @invoice_1, result: "success")
        @transaction_3 = create(:transaction, invoice: @invoice_3, result: "success")
      end
 
      it '.most revenue' do
        expect(Merchant.most_revenue(2).first.id).to equal(1)
        expect(Merchant.most_revenue(2).first.revenue.to_f).to equal(2247.46)
      end
```
 and this in the request spec:

```
      merchants = create_list(:merchant, 2)

      allow(Merchant).to receive(:most_revenue).and_return(merchants)

      get "/api/v1/merchants/most_revenue?quantity=2"

      expect(response).to be_successful

      merchant_response = JSON.parse(response.body)

      expect(merchant_response.first["id"]).to equal(1)
```

Everything is working but I want to make sure I fully understand what testing holes this leaves. My understanding is:
*  that the stub in the request spec will test that the route and json response are working as expected
* the model spec will test that the model method is returning the correct data
* The hole in the testing is that there isn't anything that run the entire process at once.

Reply:
@patrickshobe Exactly. I'm of the opinion that this is OK and preferred. That said, some devs prefer true end-to-end tests and that is a totally legit approach.

The reason I like this better is that it's super common to need to use this method in more than one place. Let's say multiple API endpoints and a couple feature tests down the road. With the stubbing/spying approach we can do the setup just one time and stub everywhere else. If the business logic for this method changes, we only need to update our test suite at the model layer. All of our features and requests probably remain as is which will save developer time and lead to a less fragile test suite. Finally, this approach leads to faster test suites since there's less writing to the database.